### PR TITLE
Implementation of Guild’s SDK widgets

### PR DESCRIPTION
**Pull request by MichalSvatos**
**First big thanks to @tazsingh for consultation and advice!**

### What changed
- use of the Guild's SDK native widgets instead of iframes
- styles of the columns

### Dev info
:exclamation: During the process I came across several problems which resulted in use of the GitHub online editor.

### Nice to have?
- load the script in `index.html` only on homepage

### CODE REVIEW

File: src/components/Splash.js [hunk 1]
```diff
@@ -144,8 +159,33 @@ export default ({
...
+      `torontojs`,
+      `UPCOMING`
+    )
+
+    window.$guild.renderGuildEventList(
+      guildCardPast,
+      `torontojs`,
+      `PAST`
+    )
+  })
```
Comment (tazsingh) [inline src/components/Splash.js:184]: We only need to do this on mount
```suggestion
  }, [])
```

File: src/components/Splash.js [hunk 2]
```diff
@@ -144,8 +159,33 @@ export default ({
...
+      guildCardUpcoming,
+      `torontojs`,
+      `UPCOMING`
+    )
+
+    window.$guild.renderGuildEventList(
+      guildCardPast,
+      `torontojs`,
+      `PAST`
+    )
```
Comment (tazsingh) [inline src/components/Splash.js:183]: All of these have return a function to run when unmounted! (This is undocumented so no way you would've known this 😅)

So, you can do this to be safe (but technically doesn't matter as this `useEffect` is only mounting once):

```suggestion
    const unmountGuildCard = window.$guild.renderGuildCard(
      guildCardRoot,
      `torontojs`
    )

    const unmountUpcomingEventList = window.$guild.renderGuildEventList(
      guildCardUpcoming,
      `torontojs`,
      `UPCOMING`
    )

    const unmountPastEventList = window.$guild.renderGuildEventList(
      guildCardPast,
      `torontojs`,
      `PAST`
    )
    
    return () => {
      unmountGuildCard()
      unmountUpcomingEventList()
      unmountPastEventList()
    }
```

File: src/components/Splash.js [hunk 3]
```diff
@@ -132,6 +133,20 @@ margin-bottom: 1rem;
 }
 `
 
+const GuildCardRootStyle = {
+  minWidth: `256px`,
+}
+
+const GuildColumnStyle = createGlobalStyle`
```
Comment (tazsingh) [inline src/components/Splash.js:140]: Would it be possible to use a `styled-component` for this instead?

```suggestion
const EventColumn = styled.div`
```

File: src/components/Splash.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // @flow
 
-import React from "react"
-import styled from "styled-components"
+import React, { useEffect } from "react"
+import styled, { createGlobalStyle } from "styled-components"
```
Comment (tazsingh) [inline src/components/Splash.js:4]: Don't need this if it's a `styled-component`

```suggestion
import styled from "styled-components"
```

File: src/components/Splash.js [hunk 5]
```diff
@@ -229,34 +269,20 @@ export default ({
...
-            loading="lazy"
-            style={{ width: `450px`, height: `360px`, border: `none`, overflow: `hidden` }}
-          />
+          <div id="guild-card-root" style={GuildCardRootStyle} />
 
         </InnerContainer>
-        <InnerContainer style={{ display: `flex`, justifyContent: `space-evenly` }}>
-          <div>
+        <InnerContainer style={{ display: `flex`, justifyContent: `space-between`, overflow: `hidden` }}>
+          <div className="container-column">
```
Comment (tazsingh) [inline src/components/Splash.js:276]: Update this to the `styled-component`:

```suggestion
          <EventColumn>
```

File: src/components/Splash.js [hunk 6]
```diff
@@ -229,34 +269,20 @@ export default ({
...
-            <iframe src="https://guild.host/embeds/guild/torontojs/events/upcoming"
-              loading="lazy"
-              scrolling="no"
-              frameBorder="0"
-              style={{ width: `400px`, minHeight: `400px`, height: '100vh', border: `none`, overflow: `hidden` }}
-            />
+            <div id="guild-card-upcoming" />
           </div>
-          <div>
+          <div className="container-column">
```
Comment (tazsingh) [inline src/components/Splash.js:280]:
```suggestion
          <EventColumn>
```

File: src/components/Splash.js [hunk 7]
```diff
@@ -229,34 +269,20 @@ export default ({
...
             <Title>Past Events</Title>
-            <iframe src="https://guild.host/embeds/guild/torontojs/events/past"
-              loading="lazy"
-              scrolling="no"
-              frameBorder="0"
-              style={{ width: `400px`, minHeight: `400px`, height: '100vh', border: `none`, overflow: `hidden` }}
-            />
+            <div id="guild-card-past" />
           </div>
```
Comment (tazsingh) [inline src/components/Splash.js:283]:
```suggestion
          </EventColumn>
```

File: src/components/Splash.js [hunk 8]
```diff
@@ -229,34 +269,20 @@ export default ({
...
+          <div className="container-column">
             <Title>Upcoming Events</Title>
-            <iframe src="https://guild.host/embeds/guild/torontojs/events/upcoming"
-              loading="lazy"
-              scrolling="no"
-              frameBorder="0"
-              style={{ width: `400px`, minHeight: `400px`, height: '100vh', border: `none`, overflow: `hidden` }}
-            />
+            <div id="guild-card-upcoming" />
           </div>
```
Comment (tazsingh) [inline src/components/Splash.js:279]:
```suggestion
          </EventColumn>
```

File: src/components/Splash.js [hunk 9]
```diff
@@ -132,6 +133,20 @@ margin-bottom: 1rem;
 }
 `
 
+const GuildCardRootStyle = {
+  minWidth: `256px`,
+}
```
Comment (tazsingh) [inline src/components/Splash.js:138]: This should be another `styled-component`:

```suggestion
const GuildCardRoot = styled.div`
  min-width: 256px;
`
```

File: src/components/Splash.js [hunk 10]
```diff
@@ -229,34 +269,20 @@ export default ({
             </ul>
           </div>
 
-          <iframe src="https://guild.host/embeds/guild/torontojs/card"
-            loading="lazy"
-            style={{ width: `450px`, height: `360px`, border: `none`, overflow: `hidden` }}
-          />
+          <div id="guild-card-root" style={GuildCardRootStyle} />
```
Comment (tazsingh) [inline src/components/Splash.js:272]: Not sure if it'll work like this:

```suggestion
          <GuildCardRoot id="guild-card-root" />
```

Or if it needs to be like this:

```suggestion
          <GuildCardRoot>
            <div id="guild-card-root" />
          </GuildCardRoot>
```

**Review by tazsingh — COMMENTED** [review]
Overall looks awesome! Thanks for putting this together 😄 

I left a bunch of nitpicky comments that should hopefully be as easy as accepting the suggestions for. But the functionality looks like it does work which is amazing! 👏 👏

File: src/components/Splash.js (hunk 1 above)
Comment (MichalSvatos) [reply src/components/Splash.js:184]: I missed that. Make sense 👍

File: src/components/Splash.js (hunk 10 above)
Comment (MichalSvatos) [reply src/components/Splash.js]: Fixed in 1359ff2

File: src/components/Splash.js (hunk 9 above)
Comment (MichalSvatos) [reply src/components/Splash.js]: Fixed in 1359ff2

File: src/components/Splash.js (hunk 8 above)
Comment (MichalSvatos) [reply src/components/Splash.js]: Fixed in 1359ff2

File: src/components/Splash.js (hunk 7 above)
Comment (MichalSvatos) [reply src/components/Splash.js]: Fixed in 1359ff2

File: src/components/Splash.js (hunk 6 above)
Comment (MichalSvatos) [reply src/components/Splash.js]: Fixed in 1359ff2

File: src/components/Splash.js (hunk 5 above)
Comment (MichalSvatos) [reply src/components/Splash.js]: Fixed in 1359ff2

File: public/index.html [hunk 11]
```diff
@@ -133,5 +133,5 @@
     Thanks!
     -->
 </body>
-
+<script src="https://guild.host/assets/0.2.0/sdk.js"></script>
```
Comment (tazsingh) [inline public/index.html:136]: Looks like this is also included above, on like 35. I can't leave a suggestion to remove that one, but I'll leave it to you to decide which one to include!

**Review by tazsingh — APPROVED** [review]
Wooo amazing work in getting this done! 🙌 

Huge improvement 👏 👏 👏

**Comment by tazsingh** [comment]
Huh why can't this be merged by @MichalSvatos ?

image


Regardless, it looks good to me. I'm merging for now but might be worth looking into the repo config in the future 👀

**Comment by usrrname** [comment]
@MichalSvatos add yourself to the website maintainer names on the volunteers.md!

**Comment by usrrname** [comment]
Thanks again for this wonderful improvement!

**Pull request merged**